osu.Desktop/DiscordRichPresence.cs

```diff
@@ -139,9 +182,25 @@ private void updateStatus()
             client.SetPresence(presence);
         }
 
+        private void onJoin(object sender, JoinMessage args)
+        {
+            game.Window?.Raise(); // users will expect to be brought back to osu! when joining a lobby from discord
+
+            if (!tryParseRoomSecret(args.Secret, out long roomId, out string? password))
+                Logger.Log("Failed to parse the room secret Discord gave us", LoggingTarget.Network, LogLevel.Error);
+
+            var request = new GetRoomRequest(roomId);
+            request.Success += room => Schedule(() =>
+            {
+                game.PresentMultiplayerMatch(room, password);
+            });
+            request.Failure += _ => Logger.Log("Couldn't find the room Discord gave us", LoggingTarget.Network, LogLevel.Error);
```

Comment:
It's "important" level at best, what if the host changed the password or ended the room by the time the player joined, I don't think an error message should pop up.
```suggestion
            if (!tryParseRoomSecret(args.Secret, out long roomId, out string? password))
                Logger.Log("Could not parse room from Discord RPC Client", LoggingTarget.Network, LogLevel.Important);

            var request = new GetRoomRequest(roomId);
            request.Success += room => Schedule(() =>
            {
                game.PresentMultiplayerMatch(room, password);
            });
            request.Failure += _ => Logger.Log($"Could not find room {roomId} from Discord RPC Client", LoggingTarget.Network, LogLevel.Important);
```

Reply:
> what if the host changed the password or ended the room

This would show in Discord since game invite messages point to the user's current presence. If you close the lobby then Discord greys out the 'join' button, and Secrets are passed from the current presence as well after you click the 'join' button so password changes also propagate.

Does "Important" show in-game? I did this mainly because it shows feedback to the user instead of "why does the game do nothing when I join a lobby?"

Reply:
Yes, will look like a "warning" message.